File: ProjetoFuncional/Itens.hs
module Itens where

data TipoItem = Ferramenta | Consumivel | Material deriving (Show, Read, Eq)
data Item = Item {nome :: String, tipoItem :: TipoItem} deriving (Show, Read, Eq)

-- Item List Consumiveis
agua = Item {nome = "Agua", tipoItem = Consumivel}
comida = Item {nome = "Comida", tipoItem = Consumivel}
kitMedico = Item {nome = "Kit Medico", tipoItem = Consumivel}
estepe = Item {nome = "Step", tipoItem = Consumivel}
combustivel = Item {nome = "Combustivel", tipoItem = Consumivel}
bateriaCarro = Item {nome = "Bateria de Carro", tipoItem = Consumivel}
rebimbocaDaParafuseta = Item {nome = "Rebimboca da Parafuseta", tipoItem = Consumivel}
carvão = Item {nome = "Carvão", tipoItem = Consumivel}


-- Item List Materiais

madeira = Item {nome = "Madeira", tipoItem = Material}
pedra = Item {nome = "Pedra", tipoItem = Material}
linha = Item {nome = "Linha", tipoItem = Material}

-- Item List Ferramenta

faca = Item {nome = "Faca de Combate", tipoItem = Ferramenta}
lanterna = Item {nome = "Lanterna", tipoItem = Ferramenta}
varaPesca = Item {nome = "Vara de Pesca", tipoItem = Ferramenta}
corda = Item {nome = "Corda", tipoItem = Ferramenta}
isqueiro = Item {nome = "Isqueiro", tipoItem = Ferramenta}
radar = Item {nome = "Radar", tipoItem = Ferramenta}
picareta = Item {nome = "Picareta", tipoItem = Ferramenta}
machado = Item {nome = "Machado", tipoItem = Ferramenta}
chapeuOperario = Item {nome = "Chapeu de Operario", tipoItem = Ferramenta}
luvas = Item {nome = "Luvas", tipoItem = Ferramenta}
bolsa = Item {nome = "Bolsa", tipoItem = Ferramenta}
garrafa = Item {nome = "Garrafa", tipoItem = Ferramenta}


counterBesta = Item {nome = "Counter Besta", tipoItem = Ferramenta}
counterArmadilha = Item {nome = "Counter Armadilha", tipoItem = Ferramenta}
counterRobo = Item {nome = "Counter Robo", tipoItem = Ferramenta}

-- Item Erro
erro = Item {nome = "Valor Invalido", tipoItem = Ferramenta}

File: ProjetoLogico/itens.pl
% ITEM = (ID, NOME, TIPO).

% Item Consumivel 

item(1, "AGUA", "CONSUMIVEL").
item(2, "COMIDA", "CONSUMIVEL").
item(3, "KIT MEDICO", "CONSUMIVEL").
item(4, "STEP", "CONSUMIVEL").
item(5, "COMBUSTIVEL", "CONSUMIVEL").
item(6, "BATERIA DE CARRO", "CONSUMIVEL").
item(7, "REBIMBOCA DA PARAFUSETA", "CONSUMIVEL").
item(8, "CARVAO", "CONSUMIVEL").

% Item Material 

item(9, "MADEIRA", "MATERIAL").
item(10, "PEDRA", "MATERIAL").
item(11, "LINHA", "MATERIAL").

% Item Ferramenta

item(12, "FACA", "FERRAMENTA").
item(13, "LANTERNA", "FERRAMENTA").
item(14, "BUSSOLA", "FERRAMENTA").
item(15, "VARA DE PESCA", "FERRAMENTA").
item(16, "CORDA", "FERRAMENTA").
item(17, "ISQUEIRO", "FERRAMENTA").
item(18, "RADAR", "FERRAMENTA").
item(19, "PICARETA", "FERRAMENTA").
item(20, "MACHADO", "FERRAMENTA").
item(21, "CHAPEU OPERARIO", "FERRAMENTA").
item(22, "MOCHILA", "FERRAMENTA").

% Receitas

% {idReceita = 1, itemA = linha, itemB = linha, resultado = corda}.
receita(1, 11, 11, 16).
% {idReceita = 2, itemA = madeira, itemB = linha, resultado = varaPesca}.
receita(2, 9, 11, 15).
% {idReceita = 3, itemA = madeira, itemB = pedra, resultado = picareta}.
receita(3, 9, 10, 19).
% {idReceita = 4, itemA = madeira, itemB = pedra, resultado = machado}.
receita(4, 9, 10, 20).